src/main/java/org/tc/controllers/CourseController.java

```diff
@@ -208,8 +234,24 @@ public ModelAndView participants(@PathVariable("courseId") int id) {
                 .getContext().getAuthentication();
         ModelAndView mav = new ModelAndView("classpath:views/participants");
         mav.addObject(HEADER_TITLE, "Course Participants");
-        mav.addObject("course", course);
+        mav.addObject(ONE_COURSE_OBJECT_NAME, course);
         mav.addObject("subscribers", courseService.getSubscribers(course));
         return mav;
     }
+
+    @RequestMapping(value = {"/mycourses"}, method = RequestMethod.GET)
+    public ModelAndView myCourses(@ModelAttribute("category") Category category) {
+        ModelAndView mav = new ModelAndView("classpath:views/mycourses");
+        mav.addObject(HEADER_TITLE, "My courses");
+        List<Course> myCourses = userService.getMyCourseList();
+        if (category.getCategoryName() != null &&
+                !category.getCategoryName().equals("All")) {
+            myCourses = myCourses.stream().filter(course -> course.getCategory()
```

Comment:
Again, bad place to do this. The maximum part of application logic should be done in services